lang/cs/Org.Apache.REEF.Tests/Functional/FaultTolerant/TestContextStart.cs

```diff
@@ -0,0 +1,199 @@
+            CleanUp(testFolder);
+            TestRun(DriverConfigurations(), typeof(ContextStartDriver), 1, "ContextStartDriver", "local", testFolder);
+            ValidateSuccessForLocalRuntime(1, testFolder: testFolder);
+
+            var messages = new List<string>();
+            messages.Add(DataDownLoadStartedMessage);
+            messages.Add(DataDownLoadCompletedMessage);
+            messages.Add(StartedHandlerMessage);
+            ValidateMessageSuccessfullyLogged(messages, "Node-*", EvaluatorStdout, testFolder, 1);
+            CleanUp(testFolder);
+        }
+
+        public IConfiguration DriverConfigurations()
+        {
+            return DriverConfiguration.ConfigurationModule
+                .Set(DriverConfiguration.OnDriverStarted, GenericType<ContextStartDriver>.Class)
+                .Set(DriverConfiguration.OnEvaluatorAllocated, GenericType<ContextStartDriver>.Class)
+                .Set(DriverConfiguration.OnContextActive, GenericType<ContextStartDriver>.Class)
+                .Build();
+        }
+
+        private sealed class ContextStartDriver :
+             IObserver<IDriverStarted>,
+             IObserver<IAllocatedEvaluator>,
+             IObserver<IActiveContext>
+        {
+            private readonly IEvaluatorRequestor _requestor;
+
+            [Inject]
+            private ContextStartDriver(IEvaluatorRequestor evaluatorRequestor)
+            {
+                _requestor = evaluatorRequestor;
+            }
+
+            public void OnNext(IDriverStarted value)
+            {
+                _requestor.Submit(_requestor.NewBuilder().Build());
+            }
+
+            public void OnNext(IActiveContext value)
+            {
+                var c = TaskConfiguration.ConfigurationModule
+                    .Set(TaskConfiguration.Identifier, "TaskID")
+                    .Set(TaskConfiguration.Task, GenericType<TestTask>.Class)
+                    .Build();
+                value.SubmitTask(c);
+            }
+
+            public void OnNext(IAllocatedEvaluator value)
+            {
+                value.SubmitContext(
+                    ContextConfiguration.ConfigurationModule
+                        .Set(ContextConfiguration.Identifier, "ContextID")
+                        .Set(ContextConfiguration.OnContextStart, GenericType<ContextStartHandler>.Class)
+                        .Build());
+            }
+
+            public void OnError(Exception error)
+            {
+                throw new NotImplementedException();
+            }
+
+            public void OnCompleted()
+            {
+                throw new NotImplementedException();
+            }
+        }
+
+        private sealed class ContextStartHandler : IObserver<IContextStart>
+        {
+            private readonly DataDownLoader _dataDownLoader;
+
+            [Inject]
+            private ContextStartHandler(DataDownLoader dataDownLoader)
```

Comment:
It is to show we can inject any object to do something with the start handler. I renamed it to make it more generic.